geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureTypeParser.scala

```diff
@@ -0,0 +1,411 @@
+/***********************************************************************
+ * Copyright (c) 2013-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.features.avro
+
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder
+import org.joda.time.format.ISODateTimeFormat
+import org.locationtech.geomesa.utils.text.{WKBUtils, WKTUtils}
+import org.locationtech.jts.geom.{Geometry, GeometryCollection, LineString, MultiLineString, MultiPoint, MultiPolygon, Point, Polygon}
+import org.opengis.feature.simple.SimpleFeatureType
+
+import java.nio.ByteBuffer
+import java.util.{Date, Locale}
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+import scala.reflect.{ClassTag, classTag}
+import scala.util.control.NonFatal
+
+object AvroSimpleFeatureTypeParser {
+  val reservedPropertyKeys: Set[String] = Set(
+    GeoMesaAvroGeomFormat.KEY,
+    GeoMesaAvroGeomType.KEY,
+    GeoMesaAvroGeomDefault.KEY,
+    GeoMesaAvroDateFormat.KEY,
+    GeoMesaAvroVisibilityField.KEY,
+    GeoMesaAvroExcludeField.KEY
+  )
+
+  /**
+   * Convert an Avro [[Schema]] into a [[SimpleFeatureType]].
+   */
+  def schemaToSft(schema: Schema): SimpleFeatureType = {
+    val builder = new SimpleFeatureTypeBuilder
+    builder.setName(schema.getName)
+
+    // any extra props on the schema go in the SFT user data
+    val sftUserData = schema.getProps
```

Comment:
Should this be documented that folks can load up userData into properties?  Also, do we have any examples / tests covering this?

Reply:
It's documented on line 36 of confluent.rst, and an example at the end of the file. The last test in AvroSimpleFeatureTypeParserTest covers it.

Reply:
Awesome!